foodsaving/groups/receivers.py

```diff
@@ -21,3 +24,40 @@ def handle_invitation_accepted(sender, **kwargs):
         'invited_at': kwargs['invited_at'].isoformat(),
         'invited_via': 'e-mail'
     })
+
+
+@receiver(post_save, sender=Group)
+def group_created(**kwargs):
+    """Ensure every group has a conversation."""
+    group = kwargs.get('instance')
+    # TODO: limit this to only run on creation
+    conversation = Conversation.objects.get_or_create_for_target(group)
+    conversation.sync_users(group.members.all())
+
+
+@receiver(pre_delete, sender=Group)
+def group_deleted(**kwargs):
+    """Delete the conversation when the group is deleted."""
+    group = kwargs.get('instance')
+    conversation = Conversation.objects.get_for_target(group)
+    if conversation:
+        conversation.delete()
+
+
+@receiver(m2m_changed, sender='groups.Group_members')
+def group_membership_change(**kwargs):
+    """Keep the conversation participants up to date with the group members."""
+
+    action = kwargs.get('action')
+    group = kwargs.get('instance')
+    user_ids = kwargs.get('pk_set')
+
+    if action == 'post_add':
```

Comment:
Yes, but it would reload all the users each time. Would work, but so does this :)